### DISCUSSION TITLE
Allow folder tree to automatically refresh

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1120] [Suggestions]
### Description

Currently when new folders get added or existing folders are renamed, the folder tree does not automatically update, you have to enter it and manually refresh it (Ctrl+R) in order to see the changes.
Auto refresh is typically the default approach that is used by all of the file managers that I have ever used / tested personally.
I can't think of any reasons/uses why it would not be auto refresh by default but if others disagree with this then this maybe could be added as an option so that everyone will be happy.

Hopefully this will be considered.
Thank You

**Comment by advice2020** [comment]
EDIT:
If added would probably have to be a setting for this to allow users to choose to have it on or off because I noticed that when you refresh the Folder Tree it also collapses all the expanded folders that were opened, all of them but the two most current being used by the left and right file pane.  I personally like this because the folder tree can get out of control with expanding tons of folders if you are navigating around a lot in left and right panes, this cleans things up with each refresh but maybe others would not desire this?

**Comment by rwperrott** [comment]
I suggest going beyond calling this a feature and rather class this as an omission bug for common-sense functionality, which can optionally be selectively-disabled when more-costly polling is required.

Most file managers can watch directories for changes using OS file system watch API(s), possibly including polling; much like text editors, like Pluma, can detect external changes to file they have opened, to prompt user if they want to reload an otherwise modified file or to keep the loaded version, even when on a mounted Samba share.
